libraries/AP_BoardConfig/IMU_heater.cpp

```diff
@@ -31,97 +31,148 @@ extern const AP_HAL::HAL& hal;
 #define HAL_HEATER_GPIO_ON 1
 #endif
 
-void AP_BoardConfig::set_imu_temp(float current)
+#if HAL_HAVE_IMU_HEATER2
+#ifndef HAL_HEATER2_GPIO_ON
+#define HAL_HEATER2_GPIO_ON 1
+#endif
+#endif
+
+void AP_BoardConfig::set_imu_temp(float current, uint8_t heater_instance)
 {
-    int8_t target = heater.imu_target_temperature.get();
-    // pass to HAL for Linux
-    hal.util->set_imu_target_temp((int8_t *)&heater.imu_target_temperature);
-    hal.util->set_imu_temp(current);
+    int8_t target;
+    if (heater_instance == 0) {
+        target = heater.imu_target_temperature.get();
+
+        // pass to HAL for Linux
+        hal.util->set_imu_target_temp((int8_t *)&heater.imu_target_temperature);
+        hal.util->set_imu_temp(current);
+
+        heater_update_and_log(heater, current, target);
+    }
+#if HAL_HAVE_IMU_HEATER2
+    else if (heater_instance == 1) {
+        target = heater2.imu_target_temperature.get();
+
+        heater_update_and_log(heater2, current, target);
+        return;
+    }
+#endif
+
+#if 0
```

Comment:
Don't really want commented code - maybe add a DEBUG define if you want to leave it in